drivers/parser/csv.go

```
@@ -0,0 +1,253 @@
+		for i := range firstRow {
+			headers = append(headers, fmt.Sprintf("column_%d", i))
+		}
+	}
+
+	// Read a few sample rows to infer data types (max 100 samples)
+	sampleRows := [][]string{}
+	maxSamples := 100
+	for i := 0; i < maxSamples; i++ {
+		row, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			logger.Warnf("Error reading sample row %d: %v", i, err)
+			break
+		}
+		sampleRows = append(sampleRows, row)
+	}
+
+	// Infer types for each column
+	for i, header := range headers {
+		dataType := inferColumnType(sampleRows, i)
+		p.stream.UpsertField(header, dataType, true) // Allow nulls by default
+	}
+
+	logger.Infof("Inferred schema with %d columns from CSV", len(headers))
+	return p.stream, nil
+}
+
+// StreamRecords reads and streams CSV records with batching and context support
+func (p *CSVParser) StreamRecords(ctx context.Context, reader io.Reader, batchSize int, callback RecordCallback) error {
+	csvReader := csv.NewReader(reader)
+	csvReader.Comma = rune(p.config.Delimiter[0])
+	if p.config.QuoteCharacter != "" {
+		csvReader.LazyQuotes = true
+	}
+
+	// Skip rows if configured
+	for i := 0; i < p.config.SkipRows; i++ {
+		_, err := csvReader.Read()
+		if err != nil {
+			return fmt.Errorf("failed to skip row %d: %w", i, err)
+		}
+	}
+
+	var headers []string
+	if p.config.HasHeader {
+		// Read header row
+		var err error
+		headers, err = csvReader.Read()
+		if err != nil {
+			return fmt.Errorf("failed to read CSV headers: %w", err)
+		}
+	} else {
+		// Generate default column names based on first row
+		firstRow, err := csvReader.Read()
+		if err != nil {
+			return fmt.Errorf("failed to read first row: %w", err)
+		}
+		for i := range firstRow {
+			headers = append(headers, fmt.Sprintf("column_%d", i))
+		}
+		// Process the first row as data
+		record := make(map[string]any)
+		for i, value := range firstRow {
+			if i < len(headers) {
+				fieldType, _ := p.stream.Schema.GetType(headers[i])
```

why are we ignoring error here?